calico.go

```diff
@@ -144,12 +149,22 @@ func cmdAdd(args *skel.CmdArgs) error {
 			// 1) Run the IPAM plugin and make sure there's an IP address returned.
 			logger.WithFields(log.Fields{"paths": os.Getenv("CNI_PATH"),
 				"type": conf.IPAM.Type}).Debug("Looking for IPAM plugin in paths")
-			result, err = ipam.ExecAdd(conf.IPAM.Type, args.StdinData)
-			logger.WithField("result", result).Info("Got result from IPAM plugin")
+			res, err := ipam.ExecAdd(conf.IPAM.Type, args.StdinData)
+			logger.WithField("result", res).Info("Got result from IPAM plugin")
 			if err != nil {
 				return err
 			}
 
+			// Convert IPAM result into current Result
+			result, err = current.NewResultFromResult(res)
+			if err != nil {
+				return err
+			}
+
+			if len(result.IPs) == 0 {
+				return goerrors.New("IPAM plugin returned missing IP config")
```

Comment:
actually, nevermind, we have another import from libcalico-go with the same import name `errors` that's why I imported Go's native errors package as `goerrors` (similar to what we do in libcalico-go)